x/lazyledgerapp/types/payformessage.go

```diff
@@ -0,0 +1,288 @@
+package types
+
+import (
+	"crypto/sha256"
+	"errors"
+	fmt "fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lazyledger/lazyledger-core/crypto/merkle"
+	"github.com/lazyledger/nmt"
+)
+
+const (
+	TypeMsgPayforMessage                   = "payformessage"
+	TypeSignedTransactionDataPayForMessage = "signedtransactiondatapayformessage"
+	ShareSize                              = 256
+)
+
+///////////////////////////////////////
+// 	MsgWirePayForMessage
+///////////////////////////////////////
+
+var _ sdk.MsgRequest = &MsgWirePayForMessage{}
+
+func (msg *MsgWirePayForMessage) Route() string { return RouterKey }
+
+func (msg *MsgWirePayForMessage) Type() string { return TypeMsgPayforMessage }
+
+// ValidateBasic checks for valid namespace length, declared message size, share
+// commitments, signatures for those share commitments, and fulfills the sdk.Msg
+// interface
+func (msg *MsgWirePayForMessage) ValidateBasic() error {
+	pubK := msg.PubKey()
+
+	// ensure that the namespace id is of length == 8
+	if len(msg.GetMessageNameSpaceId()) != 8 {
```

Comment:
For this and in other cases: at least introduce a (potentially private) constant that can be used for all occurrences of the namespace ID size. That way we can't change it at one place and forget it at another. Ideally, it would import the constants defined in lazyledger-core instead: https://github.com/lazyledger/lazyledger-core/blob/585566317e519bbb6d35d149b7e856c4c1e8657c/types/consts.go#L10